src/attribute_set.rs

```diff
@@ -227,5 +227,51 @@ macro_rules! evdev_enum {
                 self.0 as _
             }
         }
+        #[cfg(feature = "serde")]
+        #[allow(unreachable_patterns)]
+        impl serde_1::Serialize for $t {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: serde_1::ser::Serializer,
+            {
+                let value = match *self {
+                    $(Self::$c => stringify!($c),)*
+                    _ => unreachable!(),
+                };
+
+                serializer.serialize_str(value)
+            }
+        }
+        paste::paste! {
+            #[cfg(feature = "serde")]
```

Comment:
I haven't used `paste` before, but if I'm reading this right you should be able to put `#[cfg(feature = "serde")]` as an annotation just above `paste::paste! {` and remove the other ones in the block. Then you don't need to include `paste` as a dep if you're not building the serde feature.

Reply:
OK, nevermind. I tried the exact same thing yesterday and I ran into issues, and now it seems to work fine. Maybe there was just a typo in the code when i tried this. Thanks!

Reply:
you don't need this second cfg on line 247 anymore since the first should remove the entire paste section?

Actually you should be able to remove all of the cfg for serdes that are under paste.